modules/standard/Types.chpl

```diff
@@ -815,12 +815,8 @@ inline proc integral.safeCast(type T) : T where isUintType(T) {
             " with a value greater than the maximum of "+ T:string+" to "+T:string);
     }
   }
-  return this:T;
-}
 
-pragma "no doc" // documented with the other safeCast above
-inline proc integral.safeCast(type T) : T where isIntType(T) {
-  if castChecking {
+  if castChecking && isIntType(T) {
```

Comment:
If you do this with safeCast, I think you also need to include the `  compilerError("safeCast is only supported between integral types");` error below in the combined function.

Reply:
I believe this error will happen automatically because this `proc safeCast` will apply only if the argument type is integral.